pkg/lib/manifest2po.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+# Extracts translatable strings from manifest.json files.
+
+import argparse
+import json
+import os
+import re
+import sys
+
+import polib
+
+parser = argparse.ArgumentParser(
+    prog="manifest2po",
+    description="Extracts translatable strings from manifest.json files",
+)
+parser.add_argument("-d", "--directory", help="Base directory for input files")
+parser.add_argument("-o", "--output", help="Output file", required=True)
+parser.add_argument("files", nargs="+", help="One or more input files", metavar="FILE")
+
+args = parser.parse_args()
+
+entries = {}
+
+
+def process_manifest(filename, manifest):
+    if "menu" in manifest:
+        process_menu(filename, manifest["menu"])
+    if "tools" in manifest:
+        process_menu(filename, manifest["tools"])
+    if "bridges" in manifest:
+        process_bridges(filename, manifest["bridges"])
+
+
+def process_keywords(filename, keywords):
+    for v in keywords:
+        for keyword in v["matches"]:
+            push(msgid=keyword, location=(filename, "0"))
+
+
+def process_docs(filename, docs):
+    for doc in docs:
+        push(msgid=doc["label"], location=(filename, "0"))
+
+
+def process_menu(filename, menu):
+    for m in menu:
+        if "label" in menu[m]:
+            push(msgid=menu[m]["label"], location=(filename, "0"))
+        if "keywords" in menu[m]:
+            process_keywords(filename, menu[m]["keywords"])
+        if "docs" in menu[m]:
+            process_docs(filename, menu[m]["docs"])
+
+
+def process_bridges(filename, bridges):
+    for b in bridges:
+        if "label" in b:
+            push(msgid=b["label"], location=(filename, "0"))
+
+
+def push(msgid, location):
+    """
+    Push an entry onto the list
+    """
+    key = msgid
+    if key in entries:
+        entries[key].occurrences.append(location)
+    else:
+        entries[key] = polib.POEntry(msgid=msgid, occurrences=[location])
+
+
+# Now process each file in turn
+for filename in args.files:
+    if os.path.basename(filename) != "manifest.json":
+        continue
+
+    # Qualify the filename if necessary
```

Comment:
I'm also not a fan of this.  The filenames given on the command-line should definitely be relative to the cwd at the time of invocation, not to the given top-level directory.

Reply:
same comment as above ("behaviour change")